### PR TITLE
LPS-157162 Fix alert behavior on action builder

### PR DESCRIPTION
**Pull request by rebsilva**
Related Issues:

- https://issues.liferay.com/browse/LPS-157162
- https://issues.liferay.com/browse/LPS-157243

Hello Reviewer!

This is a fix for some behaviors of alerts inside the Action Builder, like the `onClose()` function that wasn't working and to only show the error alert when necessary. For more information please see the tickets above. Thanks for your time!

### CODE REVIEW

**Comment by rebsilva** [comment]
ci:test:sf

**Comment by rebsilva** [comment]
ci:test:relevant

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectAction/tabs/ActionBuilder.tsx [hunk 1]
```diff
@@ -309,23 +333,38 @@ export default function ActionBuilder({
...
+		}
+		else {
+			setErrorAlert(false);
+		}
+	}, [
+		hasEmptyValues,
+		hasSyntaxError,
+		setErrorAlert,
+		setPredefinedValuesAlertMessage,
+	]);
```
Comment (miltonmc) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectAction/tabs/ActionBuilder.tsx:359]: It no seems the case for a `useEffect()`, you probably may replace this for a `useMemo()`

**Comment by rebsilva** [comment]
Hi @miltonmc! I've made the changes you asked in https://github.com/liferay-forms/liferay-portal/pull/1811 on this PR to send everything together, hope you don't mind. Also as I said above, I didn't find  a way to transform that `useEffect` into a `useMemo`, please let me know if any change is still necessary after this. Thanks a lot!

**Comment by rebsilva** [comment]
ci:test:sf

**Comment by rebsilva** [comment]
ci:test:relevant

**Comment by rebsilva** [comment]
ci:test:sf

**Comment by rebsilva** [comment]
Hi @miltonmc! Just to let you know that I had to change some behaviors that we did together, like the `setErrors` from `useForm`; it wasn't possible because of some conflict between the order it was declared and its usage (I couldn't make this work, seemed like a dependency to me). I can explain to you better tomorrow if you want.

**Comment by rebsilva** [comment]
ci:test:relevant

**Comment by rebsilva** [comment]
ci:test:relevant

**Review by miltonmc — APPROVED** [review]
Good job!

I caught just a really small tweak to do, but feel free to send in any other PR next time.

**Comment by miltonmc** [comment]
ci:forward

**Pull request closed without merging**